ticketbuyer/purchase.go

```diff
@@ -81,14 +82,19 @@ out:
 				case <-quit:
 					return
 				}
-				// Purchase tickets for each attached block, not just for the
-				// update to the main chain.  This is probably not optimal but
-				// it matches how dcrticketbuyer worked.
-				for h := v.NewHeight - int32(len(v.AttachedBlocks)) + 1; h <= v.NewHeight; h++ {
-					p.purchase(int64(h))
+
+				defer close(s2) // defer unblocking next worker
+				blockHash := v.AttachedBlocks[len(v.AttachedBlocks)-1]
+				blockInfo, err := p.w.BlockInfo(wallet.NewBlockIdentifierFromHash(blockHash))
+				if err != nil {
+					log.Errorf("failed to get block info using block hash %s", blockHash.String())
+					return
 				}
 
-				close(s2) // unblock next worker
+				// only try buying tickets on blocks 5 minutes old or less
+				if time.Now().Unix()-blockInfo.Timestamp <= int64(p.w.ChainParams().TargetTimePerBlock) {
```

Comment:
this is now wrong because the units aren't the same.  the block timestamp is in seconds but the target time per block (a time.Duration) is nanoseconds.

Reply:
oops, missed a `.Seconds()` conversion. Rectified.